src/plugins/navigation-data/public/webcomponents/attitude-indicator.html

```diff
@@ -0,0 +1,632 @@
+        hudPosHandler: function(e) {
+          var sliderValue = this.$.hudPosSlider.immediateValue;
+          this.$.svg2.style.left = sliderValue + "%";
+        },
+
+        behaviors: [namespace('behaviors').oROVStandard],
+        setData: function (data) {
+          var self = this;
+          if(!jQuery.isEmptyObject(data)) {
+            requestAnimationFrame(function() {
+              self.render(data);
+            })
+          }
+        },
+        registerEmitterHandlers: function(emitter){
+          var self = this;
+          var lastData = null;
+
+          emitter.on('plugin.navigationData.data', function(data){
+            if(lastData == data) {return};
+            lastData = data;
+          });
+          var updateData = function(){
+            self.setData(lastData);
+            setTimeout(updateData.bind(self),50);
+          }
+          updateData();
+        },
+        attached: function () {
+          this.roll = 0;
+          this.pitch = 0;
+
+          var self = this;
+          if (this.eventEmitter !== undefined){
+            this.eventEmitter.emit('InputController.register',
+            {
+              name: 'main.toggleHeadsUpDisplay',
+              description: 'Toggle the heads-up-display on/off',
+              defaults: { keyboard: '\\' },
+              down: function () {
+                $(self.$.horizon).toggleClass('hidden');
+              }
+            });
+          }
+          this.draw();
+        },
+        render: function (data) {
+          this.setValues({
+            roll: data.roll,
+            pitch: data.pitch,
+          });
+          this.draw();
+        },
+        setValues: function setValues(values) {
+          this.roll = values.roll;
+          this.pitch = values.pitch;
+        },
+        draw: function draw() {
+          if(isDemo)
+          {
+            //Generates a sine wave
+            var sinVal = Math.sin(demoCounter);
+            demoCounter += Math.PI / 100;
+
+            this.pitch = 90*sinVal;
+            this.roll = 25*sinVal;
+
+            var yPos;
+            if(this.pitch > 90)
+            {
+              yPos = 90;
+            }
+            else if(this.pitch < -90)
+            {
+              yPos = -90;
+            }
+            else
+            {
+              yPos = map(this.pitch, -90, 90, -287, 112);
+            }
+            this.$.crosshair.style.transform = "translate3d(-117px," + yPos + "px, 0px) rotate(" + this.roll + "deg)"; 
+          }
+          else
+          {
+            var yPos;
+            if(this.pitch > 90)
+            {
+              yPos = 90;
+            }
+            else if(this.pitch < -90)
+            {
+              yPos = -90;
+            }
+            else
+            {
+              yPos = map(this.pitch, -90, 90, -287, 112);
+            }
+            this.$.crosshair.style.transform = "translate3d(-117px," + yPos + "px, 0px) rotate(" + this.roll + "deg)";
```

Comment:
This code is a little hard to ready. If you simply moved the code to a helper function mapPitchToCrossHairYPos() that would help a lot.